src/App.js

```diff
@@ -1,16 +1,23 @@
-import { Routes, Route } from "react-router-dom";
-import React from "react";
+import React, { useEffect } from "react";
+import { Routes, Route, useLocation, useSearchParams } from "react-router-dom";
```

Comment:
[Routes, Route](https://www.simplilearn.com/tutorials/reactjs-tutorial/routing-in-reactjs#:~:text=The%20Router%20in%20React%20JS,is%20sent%20to%20that%20route.): To supply the browser with an asynchronous URL that corresponds to the data that will show on the web page.

[react-router-dom](https://blog.logrocket.com/react-router-dom-tutorial-examples/#:~:text=React%20Router%20DOM%20enables%20you,of%20the%20app%20and%20platform.): Contains the DOM bindings for react React Router. In other words, the router components for websites.